core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/progress/listener/ProgressUpdater.java

```diff
@@ -0,0 +1,166 @@
+
+        ProgressSnapshot uploadProgressSnapshot = uploadProgressSnapshotBuilder.build();
+        requestBodyProgress = new DefaultSdkExchangeProgress(uploadProgressSnapshot);
+
+        DefaultProgressSnapshot.Builder downloadProgressSnapshotBuilder = DefaultProgressSnapshot.builder();
+        downloadProgressSnapshotBuilder.transferredBytes(0L);
+        Optional.ofNullable(contentLength).ifPresent(downloadProgressSnapshotBuilder::totalBytes);
+
+        ProgressSnapshot downloadProgressSnapshot = downloadProgressSnapshotBuilder.build();
+        responseBodyProgress = new DefaultSdkExchangeProgress(downloadProgressSnapshot);
+
+        context = ProgressListenerContext.builder()
+                                         .request(sdkRequest)
+                                         .uploadProgressSnapshot(uploadProgressSnapshot)
+                                         .downloadProgressSnapshot(downloadProgressSnapshot)
+                                         .build();
+
+        listenerInvoker =
+            (!sdkRequest.overrideConfiguration().isPresent() ||
+             sdkRequest.overrideConfiguration().get().progressListeners() == null)
+                          ? new ProgressListenerInvoker((Collections.emptyList()))
+                          : new ProgressListenerInvoker(sdkRequest.overrideConfiguration().get().progressListeners());
+
+        endOfStreamFuture = new CompletableFuture<>();
+    }
+
+    public SdkExchangeProgress requestBodyProgress() {
+        return requestBodyProgress;
+    }
+
+    public SdkExchangeProgress responseBodyProgress() {
+        return responseBodyProgress;
+    }
+
+    public void requestPrepared() {
+        listenerInvoker.requestPrepared(context);
+    }
+
+    public void requestHeaderSent() {
+        listenerInvoker.requestHeaderSent(context);
+    }
+
+    public void resetBytesSent() {
+        requestBodyProgress.updateAndGet(b -> b.transferredBytes(0L));
+    }
+
+    public void resetBytesReceived() {
+        responseBodyProgress.updateAndGet(b -> b.transferredBytes(0L));
+    }
+
+    public void incrementBytesSent(long numBytes) {
+        long uploadBytes = requestBodyProgress.progressSnapshot().transferredBytes();
+
+        ProgressSnapshot snapshot = requestBodyProgress.updateAndGet(b -> b.transferredBytes(uploadBytes + numBytes));
+        listenerInvoker.requestBytesSent(context.copy(b -> b.uploadProgressSnapshot(snapshot)));
+    }
+
+    public void incrementBytesReceived(long numBytes) {
+        long downloadedBytes = responseBodyProgress.progressSnapshot().transferredBytes();
+
+        ProgressSnapshot snapshot = responseBodyProgress.updateAndGet(b -> b.transferredBytes(downloadedBytes + numBytes));
+        listenerInvoker.responseBytesReceived(context.copy(b -> b.downloadProgressSnapshot(snapshot)));
+    }
+
+    public void registerCompletion(CompletableFuture<? extends SdkResponse> future) {
```

Comment:
Do we still need this method?

Reply:
Good catch, we dont. I will remove these